clang/lib/Lex/LiteralSupport.cpp

```diff
@@ -1029,6 +1030,21 @@ NumericLiteralParser::NumericLiteralParser(StringRef TokSpelling,
 
       isFloat = true;
       continue;  // Success.
+    // C++23 5.13.4 [lex.fcon]
+    case 'b':
+    case 'B':
```

Comment:
Shouldn't this be conditional in much on whether or not we're in a bfloat16-compatible mode, much like the Float16 support above?